### PR TITLE
Add a python package for barnacle

### PR DESCRIPTION
**Pull request by Ch3LL**
This is a work in progress to attempt to convert barnacle into a python module to make it easier for install and use. 

I have never setup a layout of a python package so any input would be greatly appreciated. 

This new PR now allows a user to run:

`pip install -e .` in the git clone directory of barnacle and then has access to the following commands:

`barnacle-build` (can build the docker images, either all of them or a specific one)
`barnacle-test` (will run a salt test using specified docker image)
`barnacle-shell` (will start up a shell with specified docker image)

I still need to add more to make sure all functionality is converted over, test more functionality, and i need to add docs, but wanted to get any input early on as i continue to work on this. There might also be some lint errors. will run a lint test on it tomorrow.

### CODE REVIEW

File: barnacle/build.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import docker
+import dockerpty
+import os
+import yaml
+
+def _get_dockerfile(dir):
+    '''
+    helper method to find dockerfiles in a directory
+    '''
+    docker_file = 'Dockerfile'
```
Comment (cachedout) [inline barnacle/build.py]: No need to assign this. Just use it later down.

File: barnacle/build.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+def build(conf, os=None, all=False):
+    '''
+    build docker images
+    '''
+    def _build_image(path, os):
+        for line in client.build(path=path,
+                                 tag='salt' + os,
+                                 stream=True):
+            print(line.split('":"')[1])
```
Comment (cachedout) [inline barnacle/build.py]: This might blow up if client.build fails. Better error handling might be good here.

File: barnacle/build.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    build_dir = conf['barnacle_dir']
+    client = barnacle.helper._get_client()
+
+    if all:
+        all_os = (_get_dockerfile(build_dir))
```
Comment (cachedout) [inline barnacle/build.py]: Again, I'd just not do this assignment and do it in the line below.

**Comment by Ch3LL** [comment]
@cachedout this is ready for review

File: barnacle/__init__.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import barnacle.conf
+
+
+class Barnacle(object):
+    '''
+    Barnacle Class to gather configurations
+    '''
+    def __init__(self):
+        config = '/etc/barnacle.conf'
+        self.opts = barnacle.conf.get_conf(config)
```
Comment (cachedout) [inline barnacle/__init__.py]: IMHO the previous two lines should be combined.

File: barnacle/__init__.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        parser = barnacle.parser.get_args()
+        self.args = parser.parse_args()
```
Comment (cachedout) [inline barnacle/__init__.py]: Same here.

File: barnacle/build.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+def build(conf, img_os=None, all_os=False):
+    '''
+    build docker images
+    '''
+    def _build_image(path, img_os):
+        try:
+            for line in client.build(path=path,
+                                     tag='salt-' + img_os,
+                                     stream=True):
+                print(line.split('":"')[1])
```
Comment (cachedout) [inline barnacle/build.py]: Wrapped with `if '":"' in line`?

File: barnacle/test.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+# import third party libraries
+import dockerpty
+
+def run_test(test, test_os, conf, image='salt-'):
+    '''
+    method to run a salt test with container
+    '''
+    salt_volume = conf['salt_dir']
+    docker_volume = '/testing/'
```
Comment (cachedout) [inline barnacle/test.py]: Should we move this out into the config?

**Comment by cachedout** [comment]
@Ch3LL Did you want to come back to this one? If so, there's a small merge conflict.

**Comment by Ch3LL** [comment]
@cachedout should be ready for review/merge

**Comment by Ch3LL** [comment]
ping @cachedout another reminder this is ready for review/merge

**Pull request merged**